assets/js/func.js

```diff
@@ -114,36 +140,40 @@ class Pagination {
             $(`.page-number li:nth-child(${this.current_page})`).addClass('active');
         })
     }
-    
+
     load_data(start) {
         let html = '<tr>';
         let end = parseInt(this.count_row) + parseInt(start);
-        for (let k in this.data) {
-            if (k < end && k >= start || start == -1) {
-                html += '<td>' + this.data[k].id + '</td>';
-                html += '<td><a href="' + this.data[k].link + '">' + this.data[k].name + '</a></td>';
-                html += '<td>' + this.data[k].des + '</td>';
-                html += '<td>' + this.data[k].start + '</td>';
-                html += '<td>' + this.data[k].end + '</td>';
-                html += '<td>';
-                for (let i in this.data[k].Dev) {
-                    html += '<a href="' + this.data[k].Dev[i].link + '">' + this.data[k].Dev[i].name + ' </a>';
-                }
-                html += '</td>';
-                html += '<td><div class="btn-group-sm"><button type="button" class="btn btn-primary">Update</button><button type="button" class="btn btn-danger">Remove</button></div></td>';
-                html += '</tr>';
+        let dataView;
+        console.log(start, end);
+        dataView = this.data.slice(start, end);
+        //console.log(dataView[1].id)
+        for (let k in dataView) {
+
+            html += '<td>' + dataView[k].id + '</td>';
+            html += '<td><a href="' + dataView[k].link + '">' + dataView[k].name + '</a></td>';
+            html += '<td>' + dataView[k].des + '</td>';
+            html += '<td>' + dataView[k].start + '</td>';
+            html += '<td>' + dataView[k].end + '</td>';
+            html += '<td>';
+            for (let i in dataView[k].Dev) {
+                html += '<a href="' + dataView[k].Dev[i].link + '">' + dataView[k].Dev[i].name + ' </a>';
             }
+            html += '</td>';
+            html += '<td><div class="btn-group-sm"><button type="button" class="btn btn-primary btn-func">Update</button><button type="button" class="btn btn-danger">Remove</button></div></td>';
+            html += '</tr>';
+
```

Comment:
Use the template strings instead of the join strings.

Reference docs:
https://wesbos.com/template-strings-html
https://www.w3schools.com/js/js_string_templates.asp